### ISSUE TITLE
[UX] Short white startscreen while dark theme set

### ISSUE THREAD

**Issue by peterpanondrugs** [signalapp/Signal-Android#5622]
I have:
- [x] searched open and closed issues for duplicates: Searched for: startscreen, white
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

---
### Bug description

When u start Signal from your Homescreen, 1st picture you see on your phone is a white screen at startup (approx. duration ~1s, followed by the main signal screen / your actual theme color ( in this case, dark ).
### Steps to reproduce
- in general: set dark theme in Signal
- start Signal from your homescreen

**Actual result:** Short white screen at startup while dark theme chosen.
**Expected result:** When dark theme chosen, signal should not contain any white startup screens.
### Device info

**Device:** Samsung i9305 / S3 LTE
**Android version:** 6.0.1
**Signal version:** 3.16.1
### Link to debug log

https://gist.github.com/anonymous/6b6745c542613cfed70a88a4fab7b7ac

**Comment by FeuRenard** [comment]
Some discussion already took place in #220

**Comment by peterpanondrugs** [comment]
This search engine is crap.

**Comment by pR0Ps** [comment]
Related PR (closed): #4002

**Comment by FeuRenard** [comment]
In https://github.com/WhisperSystems/Signal-Android/pull/4002#issuecomment-157873129 @mcginty says:

> An unbranded approach like https://github.com/DreaminginCodeZH/MaterialColdStart is clever, but I don't think we need to worry about it unless we need to do more time-intensive stuff at cold start than we currently do **and it also can't account for theming**.

In the referenced repository is this issue https://github.com/DreaminginCodeZH/MaterialColdStart/issues/1 where dynamic theming is addressed. It doesn't seem easy to be implemented.

**Comment by Dyras** [comment]
I really, REALLY wish this was implemented. I'm part of the light sensitive "community" of people, and I just can't stand that bright blinking at night, or even during the day for that matter! Signal is the app I use the very most, and this thing that is trivial to some is completely ruining the app for me. @moxie0 My lord and savior, I've supported every decision you've made with this app so far! Removing encrypted text feature was fine, cutting it with Cyanogenmod was great, when my IRC friends called you a douche for "kicking out" Libre from the server I stood up to you like the lapdog that I am! My eyes need this! Don't let my eyes down man! Every time my low memory Moto G3 starts Signal my ICE gets hit by FIRE!

On a more serious note, I did some "research" on my phone regarding how other apps handles the original cold startup.

Eye cancer white like Signal:
Album
Chrome
Facebook Messenger
Instagram
YouTube

Eye friendly black:
Android Sleep
Google Duo (!!)
I Can't Wake Up!
K-9
Stock Motorola Camera

tl;dr: My eyes are light sensitive and my computer monitor is set to such a strange contrast that white is gray to me. McGinty makes the argument in the previously linked pull request that black->white will hurt more than white->black, and well, I'd say the standard theme colors we have now is pretty terrible as it is and I see no reason why they couldn't be changed to something more "in-between" black and white. Come on, eye cancer white, who wants that? Not me! **Besides, can't the loading screen be blue?** You know, like the Signal icon?

(I'm not expecting a response to this, and I'm not trying to start a conversation over this. I know this isn't a discussion forum. I'm just pointing out that to some people with the kind of eye problem that I have, the white screen sucks. Thanks!)

**Comment by moxie0** [comment]
Didn't I hear something about some kind of native dark theme system API or something now?

**Comment by Dyras** [comment]
http://www.androidcentral.com/night-mode-back-and-we-hope-its-here-stay
If that's what you are talking about, it seems to only be available in the system UI tuner, which means it's probably something not worth bothering with since 99% of people don't even know it exists.
http://www.androidauthority.com/dark-mode-android-n-preview-693480/
That article mentions "dark mode" but it either seems to be the same thing, or to be available in the same place. Either way I can't try it out, because Moto G3 is still frying that Marshmallow.

**Comment by 2-4601** [comment]
That dark mode was dropped from Nougat because of performance issues.

Not sure if there's another kind of a dark theme available (that's not based on adjusting the screen or applying overlays).

**Comment by rajsinghUSA** [comment]
Any progress on this? It's actually a very painful experience.

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
